src/quail/PowerTask.cpp

```diff
@@ -0,0 +1,27 @@
+#include "PowerTask.hpp"
+#include "main.hpp"
+
+PowerTask::PowerTask(uint8_t priority) : Task(priority, "LED"){
+    evgroup = xEventGroupCreateStatic(&evbuf);
+};
+
+void PowerTask::activity()
+{
+    while(true){
+        uint16_t i_sense = adc0.read(0x03); // ADC0/AIN[2] is PB09
+        // 0.002 volts per amp
+        // x100 = 0.2 volts per amp post amplification
+        // 3.3V / 0.2 = 16.5A full scale
+        // ADC is in 10 bit mode -> divide by 1024
+        float i_batt = ((float) i_sense) / (1024 / 16.5); // amps
+        sys.statedata.setBatteryCurrentState(i_batt);
+        uint16_t v_sense = adc0.read(0x02); // ADC0/AIN[3] is PB08
+        // Using a voltage divider
+        // R2 = 43kΩ
+        // R4 = 10kΩ
+        // V_batt = (43000 + 10000) / 10000 * V_sense
+        float v_batt = ((float) v_sense) / (1024 / 3.3) * 53/10;
+        sys.statedata.setBatteryVoltageState(v_batt);
+
```

Comment:
seems like this task can run pretty fast - maybe consider adding in a little vTaskDelay so that we leave time for idle tasks

Reply:
I think like 200ms would be better. 1 second is a long time.
Also, what we really want is this to run at like 100hz or something, and then the slate dump to JSON to be like 5hz. We might have internal logic in quail that uses this data from the slate and would want that data faster than 5hz.